tests/test_filetools_txt.py

```diff
@@ -18,6 +20,41 @@ def test_check_encoding_of_file():
         assert check_encoding_of_file(PATH_TO_TEST_OUTPUT_TXT_FILE) == encoding
         PATH_TO_TEST_OUTPUT_TXT_FILE.unlink()
 
+@pytest.mark.parametrize(
+    "file_stem, line_number_to_cut, line_number_to_insert_before",
+    [
+        ("tc1-2", 1, 2),
+        ("tc1-2", 2, 2),
+        ("tc3-4", 3, 2),
+        ("tc3-4", 2, 4),
+        ("tc5", 1, 5),
+        ("tc6", 4, 1),
+        ("tc7", 3, "END"),
+        ("tc8", 3, 0),
+        ("tc9-11", 0, "END"),
+        ("tc9-11", 0, "end"),
+        ("tc12", 5, 0),
+    ],
+)
+def test_move_line(file_stem, line_number_to_cut, line_number_to_insert_before):
+    dir_with_test_files = DIR_WITH_FILETOOLS_TEST_FILES / "move_line"
+    tmp_output_file = dir_with_test_files / "output_tmp.txt"
+    move_line(
+        file=dir_with_test_files / f"{file_stem}.txt",
+        line_number_to_cut=line_number_to_cut,
+        line_number_to_insert_before=line_number_to_insert_before,
+        output_file=tmp_output_file,
+    )
+    check_existence_of_output_txt_file_and_compare_with_benchmark(
+        output_file=tmp_output_file,
+        benchmark_file=dir_with_test_files / "benchmark.txt"
+    )
+
+
+def test_move_line_throws_exception_with_wrong_string():
+    with pytest.raises(TypeError):  # In the tutorial, they also recommended to check the error message, but I didn't manage to implement it because the message may change depending on the wrong argument
```

Comment:
the `match` argument can contain a part of the error message, not necessarily the whole message. regex is used under the hood to perform the check